### PR TITLE
SignalR Get Started Update

### LINKED ISSUE #28462: SignalR with TypeScript: Update to 7.0 VS - VSC
Update this topic and app samples to .NET 7.0

There are two tabbed versions to update:

Visual Studio for Windows
Visual Studio Code

Look over all other open issues while on this to see if they can be handled:

Add a Next Steps section and point to:
[Strongly typed hubs](https://learn.microsoft.com/en-us/aspnet/core/signalr/hubs?view=aspnetcore-7.0#strongly-typed-hubs-3).

Also related issue for an update of the JavaScript version https://github.com/dotnet/AspNetCore.Docs/issues/18740

---
#### Document Details

⚠ *Do not edit this section. It is required for learn.microsoft.com ➟ GitHub issue linking.*

* ID: 47557f89-3a14-1139-0074-38d931b35faf
* Version Independent ID: 1cee43b4-b672-e480-3e26-72e08a6c2d77
* Content: [Tutorial: Get started with ASP.NET Core SignalR using TypeScript and Webpack](https://learn.microsoft.com/en-us/aspnet/core/tutorials/signalr-typescript-webpack?view=aspnetcore-7.0&tabs=visual-studio)
* Content Source: [aspnetcore/tutorials/signalr-typescript-webpack.md](https://github.com/dotnet/AspNetCore.Docs/blob/main/aspnetcore/tutorials/signalr-typescript-webpack.md)
* Product: **aspnet-core**
* Technology: **aspnetcore-tutorials**
* GitHub Login: @ssougnez
* Microsoft Alias: **wpickett**

---
[Associated WorkItem - 64678](https://dev.azure.com/msft-skilling/Content/_workitems/edit/64678)

### LINKED ISSUE #28752: 8.0 Update to includes for prereqs:  SDK, VS, VSC and VS for Mac
Provide 8.0 versions for the following prereq include files:

/includes/7.0-SDK.md
/includes/net-prereqs-vs-7.0.md
/includes/net-prereqs-vsc-7.0.md
/includes/net-prereqs-mac-7.0.md

### PR DESCRIPTION
**Pull request by wadepickett**
[Internal review](https://review.learn.microsoft.com/en-us/aspnet/core/tutorials/signalr-typescript-webpack?view=aspnetcore-8.0&branch=pr-en-us-28720&tabs=visual-studio)

Fixes #28462 
Fixes #28752

- Updated for both .NET 7 and 8.
- Moved to version by file, versions 2.1 through 8
- Update the prereq includes to v8.

Prework:  - Updated the app samples for this for version 7 and 8 on the samples repo with pr dotnet/aspnetcore.docs.samples#135

---

#### Internal previews

| 📄 File | 🔗 Preview link |
|:--|:--|
| [aspnetcore/tutorials/signalr-typescript-webpack.md](https://github.com/dotnet/AspNetCore.Docs/blob/a83847824db119c9bd0d77d91ac1c126a236657d/aspnetcore/tutorials/signalr-typescript-webpack.md) | [aspnetcore/tutorials/signalr-typescript-webpack](https://review.learn.microsoft.com/en-us/aspnet/core/tutorials/signalr-typescript-webpack?branch=pr-en-us-28720) |

### CODE REVIEW

**Comment by wadepickett** [comment]
@tdykstra, the build warning for code links are because the codes samples exist in the samples repo. However the code shows up fine in the doc build for all doc versions.

**Comment by Rick-Anderson** [comment]
@wadepickett  can you fix the build errors, it doesn't look clean to have build errors for such a long time

image

**Comment by Rick-Anderson** [comment]
@wadepickett  don't request reviews with so many build errors, it's not ready for review.

image

**Comment by Rick-Anderson** [comment]
Reverting to draft, this is not ready for review. Please fix build errors.

**Comment by wadepickett** [comment]
@Rick-Anderson,

- For the build warnings for dupe H2's as a result of multiple versions of the doc, we don't address those, correct? 
- For the code link warnings, those are because the code is not in this repo, but the samples repo.  We do not, and cannot do anything about those correct?  Or is there a different way I am to link to them?

**Comment by Rick-Anderson** [comment]
> For the build warnings for dupe H2's as a result of multiple versions of the doc, we don't address those, correct?

Incorrect. You need to fix those. You've obviously got overlapping monikers.

> For the code link warnings, those are because the code is not in this repo, but instead, in the samples repo. So it would be a false warning, right? We do not, and cannot do anything about those correct? Or is there a different way I am to link to them? The code shows in the doc build as it is supposed to but still gives the warning.

Incorrect.

**Comment by Rick-Anderson** [comment]
@wadepickett  when you split out files like this you need to check the review build side by side with the live build, and for each version. I did that and it was obvious there were problems.  It's not the reviewers responsibility to find the basic errors of your PR's.

File: aspnetcore/tutorials/signalr-typescript-webpack.md [hunk 1]
```diff
@@ -5,15 +5,15 @@ description: This tutorial provides a walkthrough of bundling and building an AS
...
+ms.custom: mvc, engagement-fy23
+ms.date: 03/17/2023
 uid: tutorials/signalr-typescript-webpack
 ---
 # Tutorial: Get started with ASP.NET Core SignalR using TypeScript and Webpack
 
 By [Sébastien Sougnez](https://twitter.com/ssougnez) and [Scott Addie](https://twitter.com/Scott_Addie)
 
-:::moniker range=">= aspnetcore-6.0"
+:::moniker range=">= aspnetcore-8.0"
```
Comment (Rick-Anderson) [inline aspnetcore/tutorials/signalr-typescript-webpack.md:16]: add old versions warning include above this line.

Comment (wadepickett) [reply aspnetcore/tutorials/signalr-typescript-webpack.md:16]: @Rick-Anderson, sorry, what is the "old versions warning include"?  What is the name of the include you are referring to?  I'm looking through our library of includes and I am not spotting what you might be referring to.

Comment (Rick-Anderson) [reply aspnetcore/tutorials/signalr-typescript-webpack.md:16]: > @Rick-Anderson, sorry, what is the "old versions warning include"? What is the name of the include you are referring to? I'm looking through our library of includes and I am not spotting what you might be referring to.
> 
> Maybe you are referring to include/signalr-in-shared-framework.md?

```
:::moniker range="< aspnetcore-6.0"
[!INCLUDE [not-latest-version](~/includes/not-latest-version.md)]
:::moniker-end
```

Comment (Rick-Anderson) [reply aspnetcore/tutorials/signalr-typescript-webpack.md:16]: Put that at the top of the file like I said when I mentioned you in another PR.

Comment (Rick-Anderson) [reply aspnetcore/tutorials/signalr-typescript-webpack.md:16]: It does [this](https://learn.microsoft.com/en-us/aspnet/core/fundamentals/host/hosted-services?branch=pr-en-us-28734&view=aspnetcore-5.0&tabs=visual-studio)

Comment (wadepickett) [reply aspnetcore/tutorials/signalr-typescript-webpack.md:16]: Thanks. Are we setting this include to show for version 7 only after 8 goes GA, or now during preview?

Comment (Rick-Anderson) [reply aspnetcore/tutorials/signalr-typescript-webpack.md:20]: If the latest released version is not selected, the warning is displayed.

**Pull request merged**